app/models/manageiq/providers/redhat/infra_manager/supported_apis_mixin.rb

```diff
@@ -0,0 +1,85 @@
+module ManageIQ::Providers::Redhat::InfraManager::SupportedApisMixin
+  def supported_api_versions
+    return supported_api_versions_from_cache if api_version.blank?
+    supported_api_versions_from_db
+  end
+
+  DEFAULT_PRODUCT_VERSION_TO_API_VERSION_REGEX_HASH = {
+    '3\.\d+\.?\d*' => %w(3),
```

Comment:
There is no need to assume that the product version is numeric, or that it has only three components. I'd suggest to use `^3\..*$`. Note the `^` and `$` anchors to make sure that you are looking for the begin and end of the string, and not for some random position inside it. Same for version 4.

Also, I don't see the modification to the `settings.yaml` file. I think we should modify it, even if we have the defaults in the code. That way users can find the setting when taking a look at the `settings.yaml` file.

Reply:
Regexps done.
I did not want to change the yaml because it is advanced stuff which i do not think a user should touch without really understanding what is going on, and in that case they should know how to add them. What do you think?

Reply:
I think it is important to have it in the `settings.yaml` file. I, for example, have no idea of how to do that. If it was there already, I could probably modify it without having to know how to add it. So my suggestion is to add it.

Reply:
Ok makes sense - I did, and I removed the default since it has no use in that case.